src/rust/scheduler/scheduler.rs

```diff
@@ -402,39 +421,82 @@ mod tests {
         Ok(())
     }
 
-    /// Tests if consecutive tasks are not assigned the same task id.
     #[test]
-    fn test_scheduler_task_ids() -> Result<()> {
+    fn from_task_id_returns_none_for_non_existing_task_id() -> Result<()> {
         let scheduler: Scheduler = Scheduler::default();
+        match scheduler.from_task_id(0) {
+            Some(_) => anyhow::bail!("from_task_id() must return None"),
+            None => {},
+        }
+        Ok(())
+    }
 
-        // Create and run a task.
-        let task: DummyTask = DummyTask::new(String::from("testing"), Box::pin(DummyCoroutine::new(0)));
+    #[test]
+    fn from_task_id_returns_correct_task_handle() -> Result<()> {
+        let scheduler: Scheduler = Scheduler::default();
+        let task: DummyTask = DummyTask::new(String::from("testing"), Box::pin(DummyCoroutine::new(42)));
         let handle: TaskHandle = match scheduler.insert(task) {
             Some(handle) => handle,
             None => anyhow::bail!("insert() failed"),
         };
-        let task_id: u64 = handle.clone().get_task_id();
-        scheduler.poll();
+        let task_id: u64 = handle.get_task_id();
+        match scheduler.from_task_id(task_id) {
+            Some(retreived_task_handle) => crate::ensure_eq!(task_id, retreived_task_handle.get_task_id()),
+            None => anyhow::bail!("from_task_id() must not return None"),
+        }
+        Ok(())
+    }
 
-        // Ensure that the first task has completed.
-        crate::ensure_eq!(handle.has_completed(), true);
+    #[test]
+    fn remove_removes_task_id_as_well() -> Result<()> {
```

Comment:
Could you add a Rust Doc comment for this test, like we have for other tests in this module?